File: kotlin/strategy/DuckSimulator/client/MallardDuck.kt
package client

import behavior.fly.FlyWithWings
import behavior.quack.Quack

class MallardDuck(duckName: String) : Duck(duckName) {

    init {
        flyBehavior = FlyWithWings(duckName)
        quackBehavior = Quack(duckName)
    }
}

File: python/strategy/DuckSimulator/client/MallardDuck.py
from client.Duck import Duck
from behavior.FlyBehavior import FlyWithWings
from behavior.QuackBehavior import Quack

class MallardDuck(Duck):
    """this is a mollard duck class, a subclass from Duck"""
    def __init__(self, duck_name):
        super().__init__(duck_name)
        self._fly_behavior = FlyWithWings(duck_name)
        self._quack_behavior = Quack(duck_name)
